openvasd/openvasd.c

```diff
@@ -123,14 +123,11 @@ openvasd_connector_new (void)
 {
   openvasd_connector_t connector;
   connector = g_malloc0 (sizeof (struct openvasd_connector));
-  if (!connector)
```

Comment:
The caller below at line 145 still looks like it's checking the return for NULL.

Also the checks there are using `conn == NULL` where they should be `*conn == NULL`.  Otherwise you're assigning to the address pointed to by conn when conn is NULL.

Or maybe only the first check has the error and the second check should come before the first.